### PR TITLE
Add channel flavor for publishing variant for Google Play

### PR DESCRIPTION
**Pull request by Goooler**
## Type of change

:white_check_mark: General change (non-breaking change that doesn't fit the below categories like copyediting)
:x: Bug fix (non-breaking change which fixes an issue)
:x: New feature (non-breaking change which adds functionality)
:x: Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Comment by SuperDragonXD** [comment]
I wouldn't recommend removing unused variants, because it might make rebasing Launcher3 more difficult.

Also, changing the CI to use a different app package name is probably not the best idea. For `release update`, I'd recommend building both `foss` and `market` build types.

**Comment by Goooler** [comment]
> I wouldn't recommend removing unused variants, because it might make rebasing Launcher3 more difficult.

Still need to rename some of them if we don't remove them, there are still some conflicts.

Sure, we can build both of them on CI.

**Comment by yasanglass** [comment]
Other than what @SuperDragonXD said, I dong think Foss and market are proper names for these. The app is Foss no matter where it's published. The new variant is just for playstore specifically and the only other place we publish is GitHub. 

Maybe it should just be play store and git flavors.

**Comment by Goooler** [comment]
I see it was published at [F-Droid](https://f-droid.org/zh_Hans/packages/ch.deletescape.lawnchair.plah/), so just call it `market`.

**Comment by yasanglass** [comment]
I think the `market` name is okay but the `foss` one can cause confusion. Maybe it can just be named `git` or `github`.

**Review by SuperDragonXD — COMMENTED** [review]
We can also use `default` as well.

**Comment by Goooler** [comment]
```diff
OLD: Lawnchair 14 Dev (faa5bdd) github release.apk (signature: V2)
NEW: Lawnchair 14 Dev (faa5bdd) market release.apk (signature: V2)

          │           compressed           │          uncompressed
          ├───────────┬───────────┬────────┼───────────┬───────────┬────────
 APK      │ old       │ new       │ diff   │ old       │ new       │ diff
──────────┼───────────┼───────────┼────────┼───────────┼───────────┼────────
      dex │   5.7 MiB │   5.7 MiB │ -165 B │  14.2 MiB │  14.2 MiB │ +316 B
     arsc │   4.4 MiB │   4.4 MiB │    0 B │   4.4 MiB │   4.4 MiB │    0 B
 manifest │   5.8 KiB │   5.9 KiB │  +34 B │  26.2 KiB │  26.5 KiB │ +384 B
      res │   2.6 MiB │   2.6 MiB │   -4 B │   7.6 MiB │   7.6 MiB │    0 B
   native │ 998.4 KiB │ 998.8 KiB │ +338 B │ 980.4 KiB │ 980.4 KiB │    0 B
    asset │  25.2 KiB │  25.2 KiB │   -6 B │ 152.8 KiB │ 152.7 KiB │   -2 B
    other │     933 B │     933 B │    0 B │     690 B │     690 B │    0 B
──────────┼───────────┼───────────┼────────┼───────────┼───────────┼────────
    total │  13.8 MiB │  13.8 MiB │ +197 B │  27.4 MiB │  27.4 MiB │ +698 B


         │          raw           │            unique
         ├────────┬────────┬──────┼────────┬────────┬─────────────
 DEX     │ old    │ new    │ diff │ old    │ new    │ diff
─────────┼────────┼────────┼──────┼────────┼────────┼─────────────
   files │      2 │      2 │    0 │        │        │
 strings │  92166 │  92167 │   +1 │  82629 │  82629 │ 0 (+20 -20)
   types │  22932 │  22932 │    0 │  20985 │  20985 │ 0 (+0 -0)
 classes │  18849 │  18849 │    0 │  18849 │  18849 │ 0 (+0 -0)
 methods │ 127951 │ 127951 │    0 │ 121643 │ 121643 │ 0 (+4 -4)
  fields │  66694 │  66694 │    0 │  64859 │  64859 │ 0 (+0 -0)


 ARSC    │ old  │ new  │ diff
─────────┼──────┼──────┼──────
 configs │  368 │  368 │  0
 entries │ 9114 │ 9114 │  0


=================
====   APK   ====
=================

     compressed     │    uncompressed    │
───────────┬────────┼───────────┬────────┤
 size      │ diff   │ size      │ diff   │ path
───────────┼────────┼───────────┼────────┼────────────────────────────────
 280.3 KiB │ +338 B │ 274.3 KiB │    0 B │ ∆ lib/arm64-v8a/libnrb.so
   2.7 MiB │  -92 B │   6.8 MiB │  +64 B │ ∆ classes2.dex
     3 MiB │  -73 B │   7.3 MiB │ +252 B │ ∆ classes.dex
   5.9 KiB │  +34 B │  26.5 KiB │ +384 B │ ∆ AndroidManifest.xml
     976 B │   -4 B │     844 B │    0 B │ ∆ assets/dexopt/baseline.profm
     454 B │   -4 B │     358 B │    0 B │ ∆ res/-9.png
   9.5 KiB │   -2 B │   9.3 KiB │   -2 B │ ∆ assets/dexopt/baseline.prof
───────────┼────────┼───────────┼────────┼────────────────────────────────
     6 MiB │ +197 B │  14.5 MiB │ +698 B │ (total)



======================
====   MANIFEST   ====
======================

              │ old              │ new
──────────────┼──────────────────┼───────────────────────────────
 package      │ app.lawnchair    │ ch.deletescape.lawnchair.plah
 version code │ 13               │ 13
 version name │ 14 Dev (faa5bdd) │ 14 Dev (faa5bdd)


@@ -5,3 +5,3 @@
     android:versionName="14 Dev (faa5bdd)"
-    package="app.lawnchair"
+    package="ch.deletescape.lawnchair.plah"
     platformBuildVersionCode="34"
@@ -132,9 +132,9 @@
   <uses-permission
-      android:name="app.lawnchair.permission.READ_SETTINGS"
+      android:name="ch.deletescape.lawnchair.plah.permission.READ_SETTINGS"
       />
   <uses-permission
-      android:name="app.lawnchair.permission.WRITE_SETTINGS"
+      android:name="ch.deletescape.lawnchair.plah.permission.WRITE_SETTINGS"
       />
   <permission
-      android:name="app.lawnchair.DYNAMIC_RECEIVER_NOT_EXPORTED_PERMISSION"
+      android:name="ch.deletescape.lawnchair.plah.DYNAMIC_RECEIVER_NOT_EXPORTED_PERMISSION"
       android:protectionLevel="0x2"
@@ -142,3 +142,3 @@
   <uses-permission
-      android:name="app.lawnchair.DYNAMIC_RECEIVER_NOT_EXPORTED_PERMISSION"
+      android:name="ch.deletescape.lawnchair.plah.DYNAMIC_RECEIVER_NOT_EXPORTED_PERMISSION"
       />
@@ -199,3 +199,3 @@
           android:name="com.android.launcher3.grid.control"
-          android:value="app.lawnchair.grid_control"
+          android:value="ch.deletescape.lawnchair.plah.grid_control"
           />
@@ -264,3 +264,3 @@
     <provider
-        android:authorities="app.lawnchair.fileprovider"
+        android:authorities="ch.deletescape.lawnchair.plah.fileprovider"
         android:exported="false"
@@ -286,9 +286,9 @@
         <action
-            android:name="app.lawnchair.bugreport.COPY"
+            android:name="ch.deletescape.lawnchair.plah.bugreport.COPY"
             />
         <action
-            android:name="app.lawnchair.bugreport.UPLOAD"
+            android:name="ch.deletescape.lawnchair.plah.bugreport.UPLOAD"
             />
         <action
-            android:name="app.lawnchair.bugreport.UPLOAD_COMPLETE"
+            android:name="ch.deletescape.lawnchair.plah.bugreport.UPLOAD_COMPLETE"
             />
@@ -350,3 +350,3 @@
     <provider
-        android:authorities="app.lawnchair"
+        android:authorities="ch.deletescape.lawnchair.plah"
         android:exported="true"
@@ -364,3 +364,3 @@
     <provider
-        android:authorities="app.lawnchair.overview.fileprovider"
+        android:authorities="ch.deletescape.lawnchair.plah.overview.fileprovider"
         android:exported="false"
@@ -469,10 +469,10 @@
     <provider
-        android:authorities="app.lawnchair.settings"
+        android:authorities="ch.deletescape.lawnchair.plah.settings"
         android:exported="true"
         android:name="com.android.launcher3.LauncherProvider"
-        android:readPermission="app.lawnchair.permission.READ_SETTINGS"
-        android:writePermission="app.lawnchair.permission.WRITE_SETTINGS"
+        android:readPermission="ch.deletescape.lawnchair.plah.permission.READ_SETTINGS"
+        android:writePermission="ch.deletescape.lawnchair.plah.permission.WRITE_SETTINGS"
         />
     <provider
-        android:authorities="app.lawnchair.grid_control"
+        android:authorities="ch.deletescape.lawnchair.plah.grid_control"
         android:exported="true"
@@ -499,3 +499,3 @@
     <provider
-        android:authorities="app.lawnchair.TestInfo"
+        android:authorities="ch.deletescape.lawnchair.plah.TestInfo"
         android:enabled="false"



=================
====   DEX   ====
=================

STRINGS:

   old   │ new   │ diff
  ───────┼───────┼─────────────
   82629 │ 82629 │ 0 (+20 -20)

  + cancelAnimations$lawnchair_lawnWithQuickstepMarketRelease
  + cancelInternal$lawnchair_lawnWithQuickstepMarketRelease(Ljava/util/Set;)V
  + ch.deletescape.lawnchair.plah
  + ch.deletescape.lawnchair.plah.BugReport
  + ch.deletescape.lawnchair.plah.bugreport.COPY
  + ch.deletescape.lawnchair.plah.bugreport.UPLOAD
  + ch.deletescape.lawnchair.plah.bugreport.UPLOAD_COMPLETE
  + ch.deletescape.lawnchair.plah.crashes
  + ch.deletescape.lawnchair.plah.fileprovider
  + ch.deletescape.lawnchair.plah.settings
  + ch.deletescape.lawnchair.plah.status
  + getAccessibilityService$lawnchair_lawnWithQuickstepMarketRelease
  + lawnWithQuickstepMarket
  + setAccessibilityService$lawnchair_lawnWithQuickstepMarketRelease
  + startInternal$lawnchair_lawnWithQuickstepMarketRelease()V
  + updateTargetViews$lawnchair_lawnWithQuickstepMarketRelease
  + ~~R8{"backend":"dex","compilation-mode":"release","desugared-library-identifiers":["com.tools.android:desugar_jdk_libs_configuration:2.0.4"],"has-checksums":false,"min-api":26,"pg-map-id":"6859410","r8-mode":"full","version":"8.2.33"}
  + Disabling ch.deletescape.lawnchair.plah, unable to run in a managed profile
  + ch.deletescape.lawnchair.plah.overview.fileprovider
  + ch.deletescape.lawnchair.plahImageActionsApi

  - app.lawnchair
  - app.lawnchair.BugReport
  - app.lawnchair.bugreport.COPY
  - app.lawnchair.bugreport.UPLOAD
  - app.lawnchair.bugreport.UPLOAD_COMPLETE
  - app.lawnchair.crashes
  - app.lawnchair.fileprovider
  - app.lawnchair.settings
  - app.lawnchair.status
  - cancelAnimations$lawnchair_lawnWithQuickstepGithubRelease
  - cancelInternal$lawnchair_lawnWithQuickstepGithubRelease(Ljava/util/Set;)V
  - getAccessibilityService$lawnchair_lawnWithQuickstepGithubRelease
  - lawnWithQuickstepGithub
  - setAccessibilityService$lawnchair_lawnWithQuickstepGithubRelease
  - startInternal$lawnchair_lawnWithQuickstepGithubRelease()V
  - updateTargetViews$lawnchair_lawnWithQuickstepGithubRelease
  - ~~R8{"backend":"dex","compilation-mode":"release","desugared-library-identifiers":["com.tools.android:desugar_jdk_libs_configuration:2.0.4"],"has-checksums":false,"min-api":26,"pg-map-id":"3bf3848","r8-mode":"full","version":"8.2.33"}
  - Disabling app.lawnchair, unable to run in a managed profile
  - app.lawnchair.overview.fileprovider
  - app.lawnchairImageActionsApi


METHODS:

   old    │ new    │ diff
  ────────┼────────┼───────────
   121643 │ 121643 │ 0 (+4 -4)

  + app.lawnchair.LawnchairApp getAccessibilityService$lawnchair_lawnWithQuickstepMarketRelease() → LawnchairAccessibilityService
  + app.lawnchair.LawnchairApp setAccessibilityService$lawnchair_lawnWithQuickstepMarketRelease(LawnchairAccessibilityService)
  + c5.a cancelAnimations$lawnchair_lawnWithQuickstepMarketRelease()
  + c5.a updateTargetViews$lawnchair_lawnWithQuickstepMarketRelease()

  - app.lawnchair.LawnchairApp getAccessibilityService$lawnchair_lawnWithQuickstepGithubRelease() → LawnchairAccessibilityService
  - app.lawnchair.LawnchairApp setAccessibilityService$lawnchair_lawnWithQuickstepGithubRelease(LawnchairAccessibilityService)
  - c5.a cancelAnimations$lawnchair_lawnWithQuickstepGithubRelease()
  - c5.a updateTargetViews$lawnchair_lawnWithQuickstepGithubRelease()
```

**Comment by Goooler** [comment]
@SuperDragonXD And we need to update the python script, it's your show time.

**Comment by SuperDragonXD** [comment]
Done. Should just work now.

**Comment by Goooler** [comment]
```
Traceback (most recent call last):
  File "/home/runner/work/lawnchair/lawnchair/ci.py", line 74, in <module>
    send_internal_notifications()
  File "/home/runner/work/lawnchair/lawnchair/ci.py", line 64, in send_internal_notifications
    send_artifact_to_telegram_chat(chat_id=telegram_ci_channel_id)
  File "/home/runner/work/lawnchair/lawnchair/ci.py", line 42, in send_artifact_to_telegram_chat
    with open(f"{full_path}/{os.listdir(full_path)[0]}", "rb") as artifact:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
IsADirectoryError: [Errno [21](https://github.com/LawnchairLauncher/lawnchair/actions/runs/7238929319/job/19720305670?pr=3861#step:6:22)] Is a directory: 'artifacts/debug-apk/lawnWithQuickstepGithub/debug'
```

**Comment by SuperDragonXD** [comment]
Actually fixed it now.

**Comment by Goooler** [comment]
Good to merge?

**Review by SuperDragonXD — APPROVED** [review]
Fine by me, although you should probably wait for other reviews to see if they have more suggestions.

**Review by yasanglass — APPROVED** [review]
Looks good to me but I am not sure if it's okay to remove those flavors or not, I have never used them myself. @suphon-t probably knows more about their use.

**Comment by Goooler** [comment]
Fixin authority in d506f23d334680497719220f22002a45c4d9df2c, cause
```
adb install Lawnchair\ 14\ Dev\ \(\#143\)\ market\ debug.apk
Performing Streamed Install
adb: failed to install Lawnchair 14 Dev (#143) market debug.apk: Failure [INSTALL_FAILED_CONFLICTING_PROVIDER: Scanning Failed.: Can't install because provider name app.lawnchair.search.LawnchairRecentSuggestionProvider (in package ch.deletescape.lawnchair.plah.debug) is already used by app.lawnchair]
```

**Review by suphon-t — APPROVED** [review]

File: build.gradle [hunk 1]
```diff
@@ -227,38 +228,28 @@ android {
...
-            isDefault true
         }
 
         withQuickstep {
             dimension "recents"
             minSdk 26
-            isDefault true
         }
 
-        withoutQuickstep {
```
Comment (yasanglass) [inline build.gradle:260]: @Goooler @suphon-t don't we need this for normal non-quickstep users eg playerstore?

**Pull request merged**